torch/_refs/__init__.py

```diff
@@ -2716,6 +2717,57 @@ def _normalize(
     return out, mean, rstd
 
 
+@register_decomposition(torch.ops.aten.native_group_norm.default, disable_meta=True)
+def native_group_norm(
+    input: Tensor,
+    weight: Optional[Tensor],
+    bias: Optional[Tensor],
+    N: int,
+    C: int,
+    HxW: int,
+    group: int,
+    eps: float,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    utils.check(
+        input.ndim >= 2,
+        lambda: "Expected at least 2 dimensions for input tensor but recieved "
+        + str(input.ndim),
+    )
+    utils.check(
+        C % group == 0,
+        lambda: "Expected number of channels in input to be divisible by num_groups, but got input of shape "
+        + str(input.shape)
+        + " and num_groups="
+        + str(group),
+    )
+    reduction_dims = [2, 3]
+    broadcast_dims = [0] + list(dim for dim in range(2, input.ndim))
+
+    input_reshaped = torch.reshape(input, [N, group, C // group, HxW])
+    out, mean, rstd = _normalize(input_reshaped, reduction_dims, eps)
+    out = out.view(input.shape)
+
+    if weight is None and bias is not None:
```

Comment:
The construction of `unsqueeze_bias` and `unsqueeze_weight` can be consolidated for readability:

```
unsqueeze_bias = None if bias is None else prims.expand_dims(bias, broadcast_dims, input.ndim)
unsqueeze_weight = None if weight is None else prims.expand_dims(weight, broadcast_dims, input.ndim)

if unsqueeze_weight is None:
  out = out + unsqueeze_bias
if unsqueeze_bias is None:
  out = out * unsqueeze_weight
else:  # both unsqueeze_weight and unsqueeze_bias are not None
  out = out * unsqueeze_weight + unsqueeze_bias
```